src/runtime/auto-prompt-manager.ts

```diff
@@ -1187,80 +1188,72 @@ export class AutoPromptManager {
     );
   }
 
-  async getTimestamps(): Promise<ActionTimestamps[]> {
+  async getTimestamps(): Promise<ActionsTimestamps> {
     const stringified = await this.storage_.get(
       StorageKeys.TIMESTAMPS,
       /* useLocalStorage */ true
     );
-    const array = stringified ? JSON.parse(stringified) : [];
-
-    return array
-      .map((t: ActionTimestamps) => {
-        const impressions = pruneTimestamps(
-          t.impressions || [],
-          TWO_WEEKS_IN_MILLIS
-        );
-        const dismissals = pruneTimestamps(
-          t.dismissals || [],
-          TWO_WEEKS_IN_MILLIS
-        );
-        const completions = pruneTimestamps(
-          t.completions || [],
-          TWO_WEEKS_IN_MILLIS
-        );
+    // TODO(justinchou): handle parsing error to not disrupt flow.
+    const map: ActionsTimestamps = stringified ? JSON.parse(stringified) : {};
+    return Object.entries(map).reduce(
+      (acc: ActionsTimestamps, [key, value]: [string, ActionTimestamps]) => {
         return {
-          action: t.action,
-          ...(impressions?.length && {impressions}),
-          ...(dismissals?.length && {dismissals}),
-          ...(completions?.length && {completions}),
+          ...acc,
+          [key]: {
+            impressions: pruneTimestamps(
+              value.impressions,
+              TWO_WEEKS_IN_MILLIS
+            ),
+            dismissals: pruneTimestamps(value.dismissals, TWO_WEEKS_IN_MILLIS),
+            completions: pruneTimestamps(
+              value.completions,
```

Comment:
Removed the check altogether discussed in https://github.com/subscriptions-project/swg-js/pull/3453#discussion_r1523232366 because of the strict typing. @oyj9109